tests/validation/validators/requiredHmsIf.spec.ts

```diff
@@ -0,0 +1,77 @@
+import { type RequiredIfCondition, requiredHmsIf } from 'sefirot/validation/validators'
+
+describe('validation/validators/requiredHmsIf', () => {
+  function getTestCasesForAllTypes() {
+    return [
+      { value: { hour: '01', minute: '02', second: '03' }, expected: { required: true, unrequired: true } },
+      { value: undefined, expected: { required: false, unrequired: false } },
+      { value: null, expected: { required: false, unrequired: false } },
+      { value: true, expected: { required: false, unrequired: false } },
+      { value: false, expected: { required: false, unrequired: false } },
+      { value: 1, expected: { required: false, unrequired: false } },
+      { value: 'abc', expected: { required: false, unrequired: false } },
+      { value: {}, expected: { required: false, unrequired: false } },
+      { value: { hour: '01' }, expected: { required: false, unrequired: false } },
+      { value: { hour: '01', minute: '02' }, expected: { required: false, unrequired: false } },
+      { value: { hour: null, minute: null, second: null }, expected: { required: false, unrequired: true } },
+      { value: { hour: '01', minute: '03', second: null }, expected: { required: false, unrequired: true } },
+      { value: [], expected: { required: false, unrequired: false } }
+    ]
+  }
+
+  function getTestCasesForGivenTypes() {
+    return [
+      { value: { hour: '23', minute: '59', second: '59' }, expected: { required: true, unrequired: true } },
+      { value: { hour: '23', minute: '59', second: '60' }, expected: { required: true, unrequired: true } },
+      { value: { hour: '23', minute: '59' }, expected: { required: true, unrequired: true } },
+      { value: { hour: '23', minute: null }, expected: { required: false, unrequired: true } },
+
+      { value: undefined, expected: { required: false, unrequired: false } },
+      { value: null, expected: { required: false, unrequired: false } },
```

Comment:
These two should be true on unrequired probably. Same for maybe `value: {}` too. 

For `value: boolean | string | number` it can be anything. I don't have hard preference there. If we are doing validation for value every where, then probably makes sense to return false here too for consistency. Otherwise if we are ignoring different types then maybe just true.

Reply:
Now everything is treated as valid if `condition` is true: 21c1168